docs/device_api.md

```diff
@@ -346,36 +357,71 @@ PUT /v2/entities/[ENTITY_ID]/attrs/ping
 
-**Important note**: don't use operations in the NGSI API with creation semantics. Otherwise, the entity/attribute
-will be created locally to ContextBroker and the command will not progress to the device (and you will need to delete
-the created entity/attribute if you want to make it to work again). Thus, the following operations *must not* be used:
+**Important note**: don't use operations in the NGSI API with creation semantics. Otherwise, the entity/attribute will be created locally to ContextBroker and the command will not progress to the device (and you will need to delete the created entity/attribute if you want to make it to work again). Thus, the following operations *must not* be used:
 
 * `POST /v2/entities`
 * `PUT /v2/entities`
 * `POST /v2/op/entites` with `actionType` `append`, `appendStrict` or `replace`
 * `POST /v1/updateContext` with `actionType` `APPEND`, `APPEND_STRICT` or `REPLACE`
 
-For HTTP devices, whose "endpoint": "http://[DEVICE_IP]:[PORT]" is declared, then your device is supposed to be
-listening for commands at that URL in a synchronous way.
+## Push commands
+
+Push commands are those that are sent to the device once the IoT Agent receives the request from the Context Broker. In order to send push commands it is needed to set the "endpoint": "http://[DEVICE_IP]:[PORT]" in the device provision. The device is supposed to be listening for commands at that URL in a synchronous way. Make sure the device endpoint is reachable by the IoT Agent.
+
+Once the command is delivered, the device should return the result of the command to the IoT Agent as the answer to the 
+HTTP request. This result will be progressed to the Context Broker where it will be stored in the "command_info" attribute. The status of the command will be stored in the "command_status" attribute (`OK` if everithing goes right). 
+
+Push commands are only valid for HTTP devices. For MQTT devices it is not needed to set the "endpoint" paramater. 
+
+## Pull commands
+
+Pull commands are those that are stored in the IoT Agent waiting to be retrieved by the devices. This kind of commands are tipically used for devices that doesn't have a public IP or the IP can not be reached. The device connects to the IoT Agent perodically to retrieve commands. In order to send pull commands you just need to ignore the "endpoint" parameter in the device provision.
+
+Once the command request is issued to the IoT agent, the command is stored waiting to be retrieved by the device. In that moment, the status of the command is "command_info": "PENDING". Once the command is retrieved by the device the status is updated to "command_info": "DELIVERED". Eventually, once the device makes the response request with the result of the command the status is updated to "command_info": "OK".
```

Comment:
> ... the status is updated to "command_info": "OK".

command_info or command_status ?

Reply:
Need to be fixed (in a new PR).

Reply:
OK. That's a mistake. To be corrected in the next pull request